crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs

```diff
@@ -75,22 +76,58 @@ pub(crate) fn verbose_decimal_constructor(checker: &mut Checker, call: &ast::Exp
             value: str_literal, ..
         }) => {
             // Parse the inner string as an integer.
-            let trimmed = str_literal.to_str().trim_whitespace();
-
+            //
+            // For reference, a string argument to `Decimal` is parsed in CPython
+            // using this regex:
+            // https://github.com/python/cpython/blob/ac556a2ad1213b8bb81372fe6fb762f5fcb076de/Lib/_pydecimal.py#L6060-L6077
+            // _after_ trimming whitespace from the string and removing all occurrences of "_".
+            let mut trimmed = Cow::from(str_literal.to_str().trim_whitespace());
+            if memchr::memchr(b'_', trimmed.as_bytes()).is_some() {
+                trimmed = Cow::from(trimmed.replace('_', ""));
```

Comment:
I'm fine not using memchr, but I do think we need to use `replace` because underscores may appear anywhere in the string. For example- `Decimal("1_____00")` is valid.